processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -821,6 +829,14 @@ public void close() throws IOException
                                       ? null
                                       : Ints.asList(dims[i]);
           forwardDimWriters.get(i).write(listToWrite);
+          if (listToWrite == null || listToWrite.isEmpty()) {
+            // empty row; add to the nullRows bitmap
+            nullRowsList.get(i).add(rowCount);
+          } else if (dimHasNullFlags.get(i) && listToWrite.size() == 1 && listToWrite.get(0) == 0) {
+            // If this dimension has the null/empty str in its dictionary, a row with a single-valued dimension
+            // that matches the null/empty str's dictionary ID should also be added to nullRowsList.
+            nullRowsList.get(i).add(rowCount);
+          }
```

Comment:
I got this code from IndexMergerV9 so it is probably an issue there too.

Although now I wonder if this check is really necessary. nullRowsList is just used to add empty rows to the inverted index for null. Those rows already containing explicit nulls should already be in the inverted index – so this might be duplicating work?